src/api/query.rs

```diff
@@ -96,117 +102,146 @@ impl QueryHelper {
             ApiError::NoAccountsMatched,
         )
     }
 
-    pub async fn query_block_info_by_hash(&self, block_hash: &BlockHash) -> ApiResult<BlockInfo> {
+    pub async fn query_block_info_by_hash(
+        &self,
+        block_id: impl IntoBlockIdentifier,
+    ) -> ApiResult<BlockInfo> {
         map_query_result(
-            self.client.clone().get_block_info(block_hash).await,
+            self.client.clone().get_block_info(block_id).await.map(|x| x.response),
             ApiError::NoBlocksMatched,
         )
     }
 
-    pub async fn query_block_summary_by_hash(
+    pub async fn query_block_item_summary(
         &self,
-        block_hash: &BlockHash,
-    ) -> ApiResult<BlockSummary> {
-        map_query_result(
-            self.client.clone().get_block_summary(block_hash).await,
-            ApiError::NoBlocksMatched,
-        )
+        block_id: impl IntoBlockIdentifier,
+    ) -> ApiResult<Vec<BlockItemSummary>> {
+        let event_stream =
+            self.client.clone().get_block_transaction_events(block_id).await.unwrap().response;
+        let events: Result<Vec<BlockItemSummary>, Status> = event_stream.try_collect().await;
+        Ok(events.map_err(RPCError::CallError)?)
+    }
+
+    pub async fn query_block_special_events(
+        &self,
+        block_id: impl IntoBlockIdentifier,
+    ) -> ApiResult<Vec<SpecialTransactionOutcome>> {
+        let event_stream =
+            self.client.clone().get_block_special_events(block_id).await.unwrap().response;
+        let events: Result<Vec<SpecialTransactionOutcome>, Status> =
+            event_stream.try_collect().await;
+        Ok(events.map_err(RPCError::CallError)?)
     }
 
-    pub async fn query_reward_status_by_hash(
+    pub async fn query_tokenomics_info(
         &self,
-        block_hash: &BlockHash,
+        block_id: impl IntoBlockIdentifier,
     ) -> ApiResult<RewardsOverview> {
         map_query_result(
-            self.client.clone().get_reward_status(block_hash).await,
+            self.client.clone().get_tokenomics_info(block_id).await.map(|x| x.response),
             ApiError::NoBlocksMatched,
         )
     }
 
+    pub async fn query_block_hash_from_height(&self, height: i64) -> ApiResult<BlockHash> {
+        if height < 0 {
+            return Err(ApiError::InvalidBlockIdentifier(
+                InvalidBlockIdentifierError::InvalidIndex(height),
+            ));
+        };
+        let block_height = BlocksAtHeightInput::Absolute {
+            height: AbsoluteBlockHeight {
+                height: height as u64,
+            },
+        };
+        let blocks = map_query_result(
+            self.client.clone().get_blocks_at_height(&block_height).await,
+            ApiError::NoBlocksMatched,
+        )?;
+        match blocks[..] {
+            [] => Err(ApiError::NoBlocksMatched),
+            // Note that unless we decide to return additional block metadata,
+            // this particular GetBlockInfo call is redundant
```

Comment:
I am not sure wht GetBlockInfo this is referring to. There isn't any here that I see.

Reply:
It was there when I moved this code out to its own block (it's from the `query_block_info` function) and I didn't take a critical look at it. I think it might refer to the fact that returning the entire `block_info` should not be necessary, but I still think it's easier to pass it around. We also need the `block_baker` field anyways. 

So far I've just removed the comment.